src/main/java/com/ably/kafka/connect/ChannelSinkMapping.java

```diff
@@ -0,0 +1,19 @@
+package com.ably.kafka.connect;
+
+import io.ably.lib.realtime.AblyRealtime;
+import io.ably.lib.realtime.Channel;
+import io.ably.lib.types.AblyException;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import javax.annotation.Nonnull;
+
+public interface ChannelSinkMapping {
+    /**
+     * Returns the channel for the given sink record.
+     *
+     * @param sinkRecord The sink record.
+     * @param ablyRealtime AblyRealtime instance.
+     * @return The channel.
+     */
+    Channel getChannel(@Nonnull SinkRecord sinkRecord,@Nonnull AblyRealtime ablyRealtime) throws AblyException;
```

Comment:
For now I only substituted it with the current implementation. But I agree that we could have a custom checked exception in addition to `AblyException`. Maybe something like `SinkRecordFailedException` with some code, message. About the later behaviour in case of exception I would personally provide a configuration property for users to skip or completely shut down the task.
We currently have this for when AblyException is thrown
https://github.com/ably/kafka-connect-ably/blob/5d9c04d09ba0c52fa9a5086b7cfda75bf634a147/src/main/java/com/ably/kafka/connect/ChannelSinkTask.java#L124-L129

Reply:
So just to explain the code above, when RetriableException is thrown the task is not shut down and the record and the record sinking is tried again, when any other exception is thrown the task will shut down completely.

Reply:
OK, that sounds reasonable to me